cloudinit/url_helper.py

```diff
@@ -447,7 +441,11 @@ def _cb(url):
     # Handle retrying ourselves since the built-in support
     # doesn't handle sleeping between tries...
     for i in count():
-        req_args["headers"] = headers_cb(url)
+        if headers_cb:
+            req_args["headers"] = headers_cb(url)
+            req_args["headers"].update(def_headers)
```

Comment:
Overriding the passed header with the default header doesn't make sense. Lets reverse that. Long term I think that this code should get cleaned up to hardcode the user-agent, but while there is still a datasource overriding the User-Agent (Vultre) lets make this more intuitive and assign def_headers first, then `.update()` with the output of `headers_cb()`.

Reply:
something like this work?
```
user_agent = "Cloud-Init/%s" % (version.version_string())
...
for i in count():
  if headers_cb:
    headers = headers_cb(url)

  if "User-Agent" not in headers:
    headers["User-Agent"] = user_agent

  req_args["headers"] = headers
```

Reply:
(and do away with the def_headers altogether)